crates/tako/src/internal/common/resources/request.rs

```diff
@@ -151,9 +151,21 @@ impl ResourceRequestVariants {
     }
 
     pub fn sort_key(&self, allocator: &ResourceAllocator) -> (f32, TimeRequest) {
-        self.variants[1..]
+        let (i, score) = self
+            .variants
+            .iter()
+            .enumerate()
+            .find_map(|(i, rq)| {
+                if allocator.is_capable_to_run(rq) {
+                    Some((i, rq.sort_key(allocator)))
+                } else {
+                    None
+                }
+            })
+            .unwrap();
```

Comment:
This makes me a bit nervous. Is there always going to be at least a single variant for which `is_capable_to_run` returns true? And why do we skip the variants before it? It would be great to at least add a comment here.

Reply:
I will add comment in the code.